exo-go/src/aws_helpers/secrets_helpers.go

```diff
@@ -0,0 +1,165 @@
+package awsHelper
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+const secretsFile string = "secrets.tfvars"
+
+// CreateSecretsStore creates an S3 bucket  and file object used for secrets management
+func CreateSecretsStore(secretsBucket, region string) error {
+	s3client := createS3client(region)
+
+	hasBucket, err := hasBucket(s3client, secretsBucket)
+	if err != nil {
+		return err
+	}
+
+	// create bucket if it doesn't exist
+	if !hasBucket {
+		_, err = s3client.CreateBucket(&s3.CreateBucketInput{Bucket: &secretsBucket})
+		if err != nil {
+			return err
+		}
+	}
+
+	// create secrets file if it doesn't exist
+	_, err = s3client.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(secretsBucket),
+		Key:    aws.String(secretsFile),
+	})
+
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			switch awsErr.Code() {
+			case s3.ErrCodeNoSuchKey:
+				return putSecretsFile(map[string]string{}, secretsBucket, region)
+			default:
+				return awsErr
+			}
+		} else {
+			return err
+		}
+	}
+
+	return err
+}
+
+// ReadSecrets reads secret key value pair from remote store
+func ReadSecrets(secretsBucket, region string) (string, error) {
+	s3client := createS3client(region)
+	err := verifyBucket(s3client, secretsBucket)
+	if err != nil {
+		return "", err
+	}
+
+	results, err := s3client.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(secretsBucket),
+		Key:    aws.String(secretsFile),
+	})
+	if err != nil {
+		return "", err
+	}
+
+	defer results.Body.Close()
```

Comment:
I don't like ignoring any errors. Can we run this at the end of the function instead of deferring it?